src/UseCases/ReadAddress/ReadAddressUseCase.php

```diff
@@ -0,0 +1,53 @@
+<?php
+
+declare( strict_types = 1 );
+
+namespace WMDE\Fundraising\AddressChangeContext\UseCases\ReadAddress;
+
+use WMDE\Fundraising\AddressChangeContext\Domain\AddressChangeRepository;
+use WMDE\Fundraising\AddressChangeContext\Domain\Model\AddressChange;
+
+class ReadAddressUseCase {
+
+	private AddressChangeRepository $addressChangeRepository;
+
+	public function __construct( AddressChangeRepository $addressChangeRepository ) {
+		$this->addressChangeRepository = $addressChangeRepository;
+	}
+
+	public function getAddressChangeByUuids( string $currentIdentifier, string $previousIdentifier ): ?AddressChangeData {
```

Comment:
While I always try to avoid nullable values (by using result DTOs or introducing a [Null Object](https://refactoring.guru/introduce-null-object)), in this case I think it's the pragmatic and best option.